packages/core/tangle/ratesetter.go

```diff
@@ -238,12 +268,76 @@ loop:
 	}
 }
 
+func (r *RateSetter) DeficitIssuerLoop() {
+	var (
+		nextBlockSize  = float64(0.0)
+		lastDeficit    = float64(0.0)
+		lastUpdateTime = float64(time.Now().Nanosecond()) / 1000000000
+	)
+loop:
+	for {
+		select {
+		// if own deficit changes, check if we can issue something
+		case excessDeficit := <-r.deficitChan:
+			r.excessDeficit.Store(excessDeficit)
+			if nextBlockSize > 0 && (excessDeficit >= nextBlockSize || r.tangle.Scheduler.ReadyBlocksCount() == 0) {
+				blk := r.issuingQueue.PopFront().(*Block)
+				r.Events.BlockIssued.Trigger(&BlockConstructedEvent{Block: blk})
+				if next := r.issuingQueue.Front(); next != nil {
+					nextBlockSize = 0.0
+				} else {
+					nextBlockSize = float64(next.Size())
+				}
+			}
+			// update the deficit growth rate estimate
+			currentTime := float64(time.Now().Nanosecond()) / 1000000000
```

Comment:
Use `time.Now().Second()` instead of dividing by 10^9

Reply:
As far as I understand, this rounds to the nearest second which would be highly inaccurate.